gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/MemberLogsMessage.java

```diff
@@ -0,0 +1,392 @@
+  //private int byteLength;
+  private Long byteLength;
+
+  /** Default constructor for deserialization. Not to be invoked directly. */
+  public MemberLogsMessage() {
+    super(true);
+  }
+
+  public MemberLogsMessage(
+      final ResultCollector<Object, Object> rc) {
+    super(rc, null, false, true);
+  }
+
+  private MemberLogsMessage(MemberExecutorMessage other) {
+    super(other);
+  }
+
+  public String getMemberId() {
+    return memberId;
+  }
+
+  public void setMemberId(String memberId) {
+    this.memberId = memberId;
+  }
+
+  public Long getOffset() {
+    return offset;
+  }
+
+  public void setOffset(Long offset) {
+    this.offset = offset;
+  }
+
+  public Long getByteLength() {
+    return byteLength;
+  }
+
+  public void setByteLength(Long byteLength) {
+    this.byteLength = byteLength;
+  }
+
+  @Override
+  protected void execute() throws Exception {
+
+    GemFireCacheImpl gemFireCache = GemFireCacheImpl.getExisting();
+    InternalDistributedSystem ids = gemFireCache.getDistributedSystem();
+    String memberId = ids.getMemberId();
+
+    Map logDetailsMap = new HashMap();
+    logDetailsMap.put("id", memberId);
+    logDetailsMap.put("name", ids.getName());
+    logDetailsMap.put("userDir", getUserDir());
+    logDetailsMap.put("logFile", getLogFile());
+    logDetailsMap.put("logData", getLog(this.offset, this.byteLength));
+
+    lastResult(logDetailsMap);
+
+  }
+
+  private String getUserDir(){
+    return System.getProperty("user.dir");
+  }
+
+  private String getLogFile() {
+    Logger rootLogger = Logger.getRootLogger();
+    Appender appender;
+    if (rootLogger != null) {
+      Enumeration<?> e = rootLogger.getAllAppenders();
+      while (e.hasMoreElements()) {
+        appender = (Appender)e.nextElement();
+        if (appender instanceof FileAppender) {
+          return ((FileAppender)appender).getFile();
+        }
+      }
+    }
+    return "";
+  }
+
+  /** Get the part of the log files given the offset and desired length of bytes */
+  private Map<String, Object> getLog(Long offset, Long byteLength) {
+
+    Map<String, Object> logsMap = new HashMap<String, Object>();
+
+    String logFile = getLogFile();
+    if (logFile == null || StringUtils.isEmpty(logFile)) {
+      logger.error("Unable to get log file.." + logFile);
+
+      logsMap.put("logText", "Unable to get log file : " + logFile);
+      logsMap.put("startIndex", 0L);
+      logsMap.put("endIndex", 0L);
+      logsMap.put("totalLength", 0L);
+
+      return logsMap;
+    }
+
+    try {
+      File logDir = new File(getUserDir());
+      File[] files = getSortedFiles(logFile, logDir);
+      Long[] fileLengths = new Long[files.length];
+      Long totalLength = 0L;
+      for(int i=0; i < files.length; i++){
+        fileLengths[i] = files[i].length();
+        totalLength += files[i].length();
+      }
+
+      // set offset if not specified
+      if(offset == null){
+        offset = totalLength - byteLength;
+      }
```

Comment:
I have added negative offset check for safe side because caller may send negative offset index which effectively be treated as zero only.
case 1 (null case) is different than zero offset. null offset basically means use default offset (which is total length less default / passed desired length of logs).